src/mono/mono/utils/mono-threads-posix.c

```diff
@@ -204,10 +208,22 @@ mono_threads_pthread_kill (MonoThreadInfo *info, int signum)
 	    result != ESRCH
 #if defined (__MACH__) && defined (ENOTSUP)
 	    && result != ENOTSUP
+#endif
+#if defined (__linux__)
+	    && result != EAGAIN
 #endif
 	    )
 		g_error ("%s: pthread_kill failed with error %d - potential kernel OOM or signal queue overflow", __func__, result);
 
+	if (result == EAGAIN && retry_count < signal_queue_ovf_retry_count) {
```

Comment:
#ifdef linux here ? to be consistent with above ?

Not sure how useful would be but personally I would add a warning here, since this can make stw take a long time